pylana/filter.py

```diff
@@ -0,0 +1,193 @@
+        'attributeName': attribute_name,
+        'min': value_min,
+        'max': value_max
+        }
+
+    return(numeric_attribute_filter)
+
+
+def create_variant_slider_filter(group_min: int, group_max: int) -> dict:
+    """
+    Create a variant slider filter to be used in a trace filter sequence.
+
+    Args:
+        group_min:
+            An integer denoting the minimum number of variant groups.
+        group_max:
+            An integer denoting the maximum number of variant groups.
+
+    Returns:
+        A dictionary containing the filter.
+    """
+    variant_slider_filter = {
+        'type': 'variantSliderFilter',
+        'min': group_min,
+        'max': group_max
+        }
+
+    return(variant_slider_filter)
+
+
+def create_activity_filter(activity: str, include: bool = True) -> dict:
+    """
+    Create an activity filter to be used in a trace filter sequence.
+
+    Args:
+        activity:
+            A string denoting the activity to filter.
+        include:
+            A boolean denoting if the activity should be included or excluded.
+
+    Returns:
+        A dictionary containing the filter.
+    """
+    activity_filter = {
+        'type': 'activityFilter',
+        'activity': activity,
+        'inverted': not include
+        }
+
+    return(activity_filter)
+
+
+def create_activity_filters(include: list, exclude: list = []) -> list:
+    """
+    Create a list of activity filters to be used in a trace filter sequence.
+
+    Args:
+        include:
+            A list of strings denoting the activities to include.
+        exclude:
+            A list of strings denoting the activities to exclude.
+
+    Returns:
+        A list containing the activity filters.
+    """
+    activity_filters = [create_activity_filter(activity) for activity in include] +\
+    [create_activity_filter(activity, include = False) for activity in exclude]
+
+    return(activity_filters)
+
+
+def create_follower_filter(pre: str, succ: str, direct_follower = False, include = True) -> dict:
+    """
+    Create a follower filter to be used in a trace filter sequence.
+
+    Args:
+        pre:
+            A string denoting the predecessor activity of the follower relation.
+        succ:
+            A string denoting the successor activity of the follower relation.
+        direct_follower:
+            A boolean denoting if the activities have to directly
+            follow each other.
+        include:
+            A boolean denoting if the follower relation should be included
+            or excluded.
+
+    Returns:
+        A list containing the activity filters.
+    """
+    if pre == 'Start':
+        pre = '__LANA_START__'
+
+    if succ == 'End':
+        succ = '__LANA_END__'
+
+    follower_filter = {
+        'type': 'followerFilter',
+        'pre': pre,
+        'succ': succ,
+        'direct': direct_follower,
+        'inverted': not include
+        }
+
+    return(follower_filter)
```

Comment:
This can be done without the `if` by using a dictionary:

```python
mapping = {'Start': '__LANA_START__', 'End': '__LANA_END__'}

return  follower_filter = {
        'type': 'followerFilter',
        'pre': mapping.get(pre, pre),
        'succ': mapping.get(succ, succ),
        'direct': direct_follower,
        'inverted': not include
        }
```

It also has the charm that we don't re-set the variables `pre` and `succ`.

Reply:
implemented